packages/strapi-plugin-settings-manager/controllers/SettingsManager.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs');
+
+module.exports = {
+  menu: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+
+    ctx.send(Service.menu);
+  },
+
+  environments: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+
+    ctx.send({ environments: Service.getEnvironments() });
+  },
+
+  languages: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+
+    ctx.send({ languages: Service.getLanguages() });
+  },
+
+  databases: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+    const { env } = ctx.params;
+
+    if (!env || _.isEmpty(_.find(Service.getEnvironments(), { name: env }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.environment.unknow' }] }]);
+
+    ctx.send({ databases: Service.getDatabases(env) });
+  },
+
+  database: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+    const { name, env } = ctx.params;
+
+    if (!env || _.isEmpty(_.find(Service.getEnvironments(), { name: env }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.environment.unknow' }] }]);
+    if (!name || _.isEmpty(_.find(Service.getDatabases(env), { name }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.database.unknow' }] }]);
+
+    const model = Service.databases(name, env);
+
+    ctx.send(model);
+  },
+
+  databaseModel: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+    const { env } = ctx.params;
+
+    const model = Service.databases('${name}', env);
+
+    ctx.send(model);
+  },
+
+  get: async ctx => {
+    const Service = strapi.plugins['settings-manager'].services.settingsmanager;
+    const { slug, env } = ctx.params;
+
+    if (env && _.isEmpty(_.find(Service.getEnvironments(), { name: env }))) return ctx.badData(null, [{ messages: [{ id: 'request.error.environment.unknow' }] }]);
+
+    const model = _.has(Service, slug) ? Service[slug](env) : undefined;
```

Comment:
```
if (!_.has(Service, slug)) {
  return ctx.badData(null, [{ messages: [{ id: 'request.error.config' }] }]);
}

ctx.send(Service[slug](env));
```

It seems more logical because you have two conditions to check the same thing.